src/main/java/org/opensearch/knn/plugin/transport/UpdateBlockedModelTransportAction.java

```diff
@@ -0,0 +1,154 @@
+ */
+public class UpdateBlockedModelTransportAction extends TransportMasterNodeAction<UpdateBlockedModelRequest, AcknowledgedResponse> {
+
+    public static Logger logger = LogManager.getLogger(UpdateBlockedModelTransportAction.class);
+
+    private UpdateBlockedModelExecutor updateBlockedModelExecutor;
+
+    @Inject
+    public UpdateBlockedModelTransportAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(
+            UpdateBlockedModelAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            UpdateBlockedModelRequest::new,
+            indexNameExpressionResolver
+        );
+        this.updateBlockedModelExecutor = new UpdateBlockedModelExecutor();
+    }
+
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected AcknowledgedResponse read(StreamInput streamInput) throws IOException {
+        return new AcknowledgedResponse(streamInput);
+    }
+
+    @Override
+    protected void masterOperation(
+        UpdateBlockedModelRequest request,
+        ClusterState clusterState,
+        ActionListener<AcknowledgedResponse> actionListener
+    ) {
+        // ClusterManager updates blocked modelIds list based on request parameters
+        clusterService.submitStateUpdateTask(
+            PLUGIN_NAME,
+            new UpdateBlockedModelTask(request.getModelId(), request.isRemoveRequest()),
+            ClusterStateTaskConfig.build(Priority.NORMAL),
+            updateBlockedModelExecutor,
+            new ClusterStateTaskListener() {
+                @Override
+                public void onFailure(String s, Exception e) {
+                    actionListener.onFailure(e);
+                }
+
+                @Override
+                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+                    actionListener.onResponse(new AcknowledgedResponse(true));
+                }
+            }
+        );
+    }
+
+    @Override
+    protected ClusterBlockException checkBlock(UpdateBlockedModelRequest request, ClusterState clusterState) {
+        return null;
+    }
+
+    /**
+     * UpdateBlockedModelTask is used to provide the executor with the information it needs to perform its task
+     */
+    private static class UpdateBlockedModelTask {
+
+        private String modelId;
+        private boolean isRemoveRequest;
+
+        /**
+         * Constructor
+         *
+         * @param modelId id of model
+         * @param isRemoveRequest should this modelId be removed
+         */
+        UpdateBlockedModelTask(String modelId, boolean isRemoveRequest) {
```

Comment:
You can use Lombok's AllAgrsConstructor annotation here

Reply:
updated